### PR TITLE
Exit component process when session key is deleted (#21658)

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #21599

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/util/sessionutil/session_util.go [hunk 1]
```diff
@@ -704,28 +713,75 @@ func (w *sessionWatcher) handleWatchErr(err error) error {
...
-			if callback != nil {
-				go callback()
-			}
-			return
-		case <-ctx.Done():
-			log.Debug("liveness exits due to context done")
-			// cancel the etcd keepAlive context
-			if s.keepAliveCancel != nil {
-				s.keepAliveCancel()
+	getResp, err := s.etcdCli.Get(context.Background(), s.getSessionKey())
```
Comment (czs007) [inline internal/util/sessionutil/session_util.go:716]: wait for a pick from 2.2.0

**Pull request merged**